Reorganize preferences layout

NSExceptional opened the pull request:
This PR makes the changes depicted in the screenshots below, fixes some bugs I mistakenly left in my previous PR, and adds the `TARGET` variable to the Makefiles which silences some warnings by setting the minimum deployment target to iOS 13.

Please review the differences in the screenshots thoroughly before you accept this PR! I adjusted the text of several labels and descriptions.

<img width="50%" alt="Root preferences screen" src="https://user-images.githubusercontent.com/8371943/119173011-3bd3dd80-ba2c-11eb-8fa3-5ee9d14cde12.jpeg">
<img width="50%" alt="New filter screen" src="https://user-images.githubusercontent.com/8371943/119173003-3aa2b080-ba2c-11eb-940f-ddb4cd5a1a26.jpeg">

NSExceptional committed:
Reorganize preferences look

- Not every section needs a header
- Sections should have long footers instead of long headers
- Group some sections together logically
- Remove redundant text from strings
- Revise some strings and section headers to be more clear

NSExceptional committed:
Fix remaining cell sizes 50 → 44

NSExceptional committed:
Fix date picker bug from last PR

eclair4151 commented:
Cool thanks! i dont think i want to block it to ios 13 though. you can compile with GO_EASY_ON_ME = 1 if it is giving you warnings because of that view control i create. i havent found a way around that yet but dont want to drop ios 11-12 support

eclair4151 commented:
Cool thanks! i dont think i want to block it to ios 13 though. you can compile with GO_EASY_ON_ME = 1 if it is giving you warnings because of that view control i create. i havent found a way around that yet but dont want to drop ios 11-12 support

eclair4151 commented:
Cool thanks! i dont think i want to block it to ios 13 though. you can compile with GO_EASY_ON_ME = 1 if it is giving you warnings because of that view control i create. i havent found a way around that yet but dont want to drop ios 11-12 support

NSExceptional committed:
Add TARGET to makefiles, iOS 11 deployment target

Also use wildcard expression in notiblockpref Makefile instead of listing out each individual file, and use @available instead of UIDevice for version checking

NSExceptional commented:
Fixed! I updated your code to use `@available(iOS 13.0, *)` instead of those macros you had. This silences the warning. Also, I made the notiblockprefs Makefile use a `*.m` wildcard expression instead of listing out all the files one by one

eclair4151 commented:
awesome thanks, ill look through this and test on monday. I did try @available before but for some reason i was unable to compile the project with some error. but i think that's just some issue with my theos setup i might need to fix.

NSExceptional commented:
Let me know what the error is if you're still getting it and I'll try and help

eclair4151 commented:
Hm i dont know what i changed but all of a sudden i am not able to build at all. Something funky with Applist using OSSpinLock. Tried googling around but didnt see any obvious fixes. any suiggestions?
(not related to your changes)

==> Compiling NBPAppChooserViewController.m (arm64)…
In file included from NBPAppChooserViewController.m:12:
In file included from /var/theos/vendor/include/AppList/AppList.h:1:
/var/theos/vendor/include/AppList/ALApplicationList.h:14:2: error: 'OSSpinLock' is deprecated: first deprecated in iOS 10.0 - Use os_unfair_lock() from <os/lock.h> instead [-Werror,-Wdeprecated-declarations]
        OSSpinLock spinLock;
        ^
/var/theos/sdks/iPhoneOS13.7.sdk/usr/include/libkern/OSSpinLockDeprecated.h:79:17: note: 'OSSpinLock' has been explicitly marked deprecated here
typedef int32_t OSSpinLock OSSPINLOCK_DEPRECATED_REPLACE_WITH(os_unfair_lock);
                ^
1 error generated.
make[4]: *** [/var/theos/makefiles/instance/rules.mk:193: /var/mobile/Development/NotiBlock/.theos/obj/debug/arm64/NBPAppChooserViewController.m.d72e45c6.o] Error 1
==> Compiling NBPAppChooserViewController.m (arm64e)…
In file included from NBPAppChooserViewController.m:12:
In file included from /var/theos/vendor/include/AppList/AppList.h:1:
/var/theos/vendor/include/AppList/ALApplicationList.h:14:2: error: 'OSSpinLock' is deprecated: first deprecated in iOS 10.0 - Use os_unfair_lock() from <os/lock.h> instead [-Werror,-Wdeprecated-declarations]
        OSSpinLock spinLock;
        ^
/var/theos/sdks/iPhoneOS13.7.sdk/usr/include/libkern/OSSpinLockDeprecated.h:79:17: note: 'OSSpinLock' has been explicitly marked deprecated here
typedef int32_t OSSpinLock OSSPINLOCK_DEPRECATED_REPLACE_WITH(os_unfair_lock);
                ^
1 error generated.

eclair4151 commented:
But addition to that, if i use GO_EASY_ON_ME to bypass that warning, i still get the error around @available

==> Linking bundle NotiBlockPref (arm64e)…
Undefined symbols for architecture arm64e:
  "___isOSVersionAtLeast", referenced from:
      -[NBPRootTableViewController addButtonTapped:] in NBPRootTableViewController.m.46d3dc29.o
      -[NBPRootTableViewController tableView:didSelectRowAtIndexPath:] in NBPRootTableViewController.m.46d3dc29.o
ld: symbol(s) not found for architecture arm64e
Not signing file
clang-10: error: linker command failed with exit code 1 (use -v to see invocation)
make[4]: *** [/var/theos/makefiles/instance/bundle.mk:37: /var/mobile/Development/NotiBlock/.theos/obj/debug/arm64e/NotiBlockPref.bundle/NotiBlockPref] Error 1
==> Linking bundle NotiBlockPref (arm64)…
Undefined symbols for architecture arm64:
  "___isOSVersionAtLeast", referenced from:
      -[NBPRootTableViewController addButtonTapped:] in NBPRootTableViewController.m.c43d26f8.o
      -[NBPRootTableViewController tableView:didSelectRowAtIndexPath:] in NBPRootTableViewController.m.c43d26f8.o
ld: symbol(s) not found for architecture arm64
Not signing file
clang-10: error: linker command failed with exit code 1 (use -v to see invocation)

NSExceptional commented:
That's weird, I have the same header with the spinlock thing in it. I'm not compiling with any special flags. Try using the `14.4` SDK from [here](https://github.com/theos/sdks)

As for the second error… which platform are you compiling this on?? O_o If you're on macOS, which version of Xcode do you have installed?

Related: https://github.com/theos/theos/issues/380

eclair4151 commented:
I have already tried the ios 14.4 headers they did not fix the issue. Additionally, i am using my iOS device to compile the tweak. For the link you sent, i had already seen that, you can see the last comment is from me :p although that fix doesnt work for me anymore

NSExceptional commented:
Do you have a mac? Not sure what else to do here; this is the correct way to gate availability, and it appears the iOS toolchain doesn't support it somehow

Also, using the actual 14.4 SDK is not the same as using the "headers", did you actually use the 14.4 SDK?

eclair4151 commented:
Sorry for the delay, been caught up. I do have a mac, i will try to compile it on ther later today. I agree with you, it totally should be the way you have it, i just need to fix my issues so I can compile the build :p

NSExceptional commented:
Gotcha, lemme know if you have the same issue on your Mac! If you do then something is just wrong with your setup and we can work it out.

eclair4151 commented:
Good news, compiling on my mac still gives my the OSSpinLock error but at least if i call go easy on me, it doesnt have an issue with the @Available and compiles fine. i am doing some testing and review now and will let you know if i have any issues/questions

eclair4151 commented:
Works great from my test. Thanks for the help!
